src/ClpIrV1Decoder.cpp

```diff
@@ -0,0 +1,224 @@
+#include "ClpIrV1Decoder.hpp"
+
+#include <emscripten/em_asm.h>
+#include <emscripten/val.h>
+
+#include <clp/ErrorCode.hpp>
+#include <clp/ffi/ir_stream/decoding_methods.hpp>
+#include <clp/ir/LogEvent.hpp>
+#include <clp/ir/LogEventDeserializer.hpp>
+#include <clp/ir/types.hpp>
+#include <clp/streaming_compression/zstd/Decompressor.hpp>
+#include <clp/TimestampPattern.hpp>
+#include <clp/TraceableException.hpp>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <span>
+#include <system_error>
+#include <utility>
```

Comment:
Let's config clang-format to assign different levels for headers like this: https://github.com/y-scope/clp-ffi-py/blob/main/.clang-format#L71